### DISCUSSION TITLE
Week 3 Discussions

### DISCUSSION THREAD

**Discussion by gberl001** [gberl001/SP24-ENR261-180#4] [General]
Ask any questions up to and including week three material

**Comment by grahamswilczewskiMCC** [comment]
For Tutorial_3_4, the switch statement at the end isn't working how it's showing for me. For case 1 and case 2, it seems like the tutorial is saying if i input 1 or 2 I'll get the output under case 2; however, when I input 1, I don't get any output at all. I was wondering what the problem is here.

**Reply by joneal823** [reply]
I was noticing this as well. I may be missing something/ be incorrect, but per the help page for these statements (https://www.mathworks.com/help/matlab/ref/switch.html), it seems that a case being fulfilled causes MATLAB to not execute any of the other case statements (thus skipping over the fprintf under case 2). It seems like we need either {1,2} or 1|2 to allow for multiple conditions here.

**Reply by gberl001** [reply]
Interesting, I had not noticed this no longer works. I'm not sure when it changed. This is generally the way to perform multiple cases in other languages although it appears this either never was the case for MATLAB or it changed at some point. 

Looking at the doc, it looks like @joneal823 is correct, you need to use {1,2} but this is a cell array and we do not learn about this until much later in the course.

Thanks for catching this and bringing it up. I'll put in an issue to fix this for future classes.
